Assets/Scripts/Inventory/IItemWithSupply.cs

```diff
@@ -0,0 +1,9 @@
+﻿namespace Inventory
+{
+    public interface IItemWithSupply
+    {
+        void DecreaseSupply();
+        void IncreaseSupply(float amount);
+        float GetRemainingSupplyPercentage();
```

Comment:
I feel that one `GetRemainingSupply()` would be a good addition too

Reply:
It would, but I had no use for it so it would just be useless code. If someone needs it in the future, they can add it quite easily.